### ISSUE TITLE
Help! Just want to remove one item...

### ISSUE THREAD

**Issue by dhughes72** [lesnitsky/flutter_localstorage#45]
Any help is super appreciated :)

I'm trying to remove a single item from the list, but they are all tagged with the String 'todos' (I'm using the example). It all works very well (thank you so much!) but no matter what I try I can't find a way to remove one item from storage. i located a way online (https://inducesmile.com/google-flutter/how-to-delete-listview-item-using-popupmenubutton-in-flutter/#inline_content) to add a little 'delete' pop up, and it does remove the item from the list properly; but when i go out and come back, the reload pulls everything from storage including the deleted item... presumably because the list is built with 'getItem('todos')'.

Here's two snippets of my current code:
    _addItem(String title) {
      setState(() {
        final item = new TodoItem(title: title, done: false);
        //code adds item to list
        list.items.add(item);
        //code adds item to storage
        _saveToStorage();
      });
    }
    //this removes the item from the list then removes from storage
    _removeItemfromList(dynamic val) {
        setState(() => list.items.removeWhere((item) => item == val));
        _deleteFromStorage();
      }

    //this puts the number into local storage; generally called in another function
    _saveToStorage() {
      //String rando = randomString(15);
      //String uhString = "hello, " + rando; 
      //sets each item with string todos, list is built looking for this word
      storage.setItem('todos', list.toJSONEncodable());
      print(list.toJSONEncodable());
    }

    //this removes the number from local storage; generally called in another function
    _deleteFromStorage() {
            storage.deleteItem();
    }

and then below is the Container where the list is built:
body: Container(
            padding: EdgeInsets.all(10.0),
            constraints: BoxConstraints.expand(),
            child: FutureBuilder(
              future: storage.ready,
              //these options build the list from storage
              builder: (BuildContext context, AsyncSnapshot snapshot) {
                if (snapshot.data == null) {
                  return Center(
                    child: CircularProgressIndicator(),
                  );
                }
  
                if (!initialized) {
                  var items = storage.getItem('todos');
  
                  if (items != null) {
                    list.items = List<TodoItem>.from(
                      (items as List).map(
                        (item) => TodoItem(
                          title: item['title'],
                          done: item['done'],
                        ),
                      ),
                    );
                  }
                  initialized = true;
                }
                // this appears to make list visible (maybe add also)
                List<Widget> widgets = list.items.map((item) {
                  return MaterialApp(
                    home: Card(
                      child: ListTile(
                        //value: item.done,
                        title: Text(item.title),
                        selected: item.done,
                        trailing: PopupMenuButton(
                            elevation: 10,
                            offset:Offset(-500, 100),
                            onSelected: _removeItemfromList,
                            icon: Icon(Icons.menu),
                            itemBuilder: (context) => [
                              PopupMenuItem(
                                height: 20,
                                value: item, 
                                child: Text("Delete"),
                          ),
                        ],
                  ),
                      )
                    )
                  );
                    }).toList();
  
                return Column(
                  children: <Widget>[
                    Expanded(
                      flex: 1,
                      child: ListView(
                        children: widgets,
                        itemExtent: 50.0,
                      ),
                    ),
                    ListTile(
                      //title: TextField(
                      //  controller: controller,
                      //  decoration: InputDecoration(
                      //    labelText: 'What to do?',
                      //  ),
                      //  onEditingComplete: _save,
                      // ),
                      trailing: Row(
                        mainAxisSize: MainAxisSize.min,
                        children: <Widget>[
                          IconButton(
                            iconSize: 40,
                            icon: Text('ADD', style: greyStyle,),
                            onPressed: ()=> _onTap(0),
                          ),
                          IconButton(
                            iconSize: 70,
                            icon: Text('CLEAR', style: greyStyle,),
                            onPressed: _clearStorage,
                            tooltip: 'Clear storage',
                          )
                        ],
                      ),
                    ),
                  ],
                );
              },
            )),

**Comment by dhughes72** [comment]
nevermind, thank you

**Comment by lesnitsky** [comment]
did you solve your issue?

**Comment by dhughes72** [comment]
hello!

I didn't solve this issue within your plugin, I instead switched over to
the 'flutter_secure_storage 3.3.3' plugin. I couldn't figure it out at all
and the new plugin helped me sort out what I needed.

I didn't want to post in the thread that I went to a different plugin
because I did find yours very easy to use and it worked very well overall.
I also didn't want to bother you with additional requests asking for help,
so when I got the new one working this morning I decided it was best to
close the request.

thanks!
